# Analyse de base des fichiers concernant les lycées


Objectif : faire un fichier pour lancer des analyses.

In [1]:
# import bibliothéques d'analyse de python
import pandas as pd
import numpy as np


In [2]:
ips_lycee = pd.read_csv('donnee/fr-en-ips-lycees-ap2022.csv',sep=';')
va_lycee_gen = pd.read_csv('donnee/fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique.csv',sep=';',low_memory=False)
va_lycee_pro = pd.read_csv('donnee/fr-en-indicateurs-de-resultat-des-lycees-denseignement-professionnels.csv',sep=';',low_memory=False)

## Affichage des fichiers bruts

In [3]:
ips_lycee.head(1)

Rentrée scolaire Académie Code du département Département       UAI  \
0        2022-2023     LYON                 001         AIN  0010021T   

                               Nom de l'établissment Code INSEE de la commune  \
0  LYCEE PROFESSIONNEL MARCELLE PARDE LYCEE DES M...                    01053   

  Nom de la commune Secteur Type de lycée  Effectifs voie GT  \
0   BOURG EN BRESSE  public            LP                NaN   

   Effectifs voie PRO  Effectifs Ensemble GT-PRO  IPS voie GT  IPS voie PRO  \
0               520.0                      520.0          NaN          86.5   

   IPS Ensemble GT-PRO  Ecart-type de l'IPS voie GT  \
0                 86.5                          NaN   

   Ecart-type de l'IPS voie PRO  
0                          24.3

In [4]:
ips_lycee.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3598 entries, 0 to 3597
Data columns (total 18 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Rentrée scolaire              3598 non-null   object 
 1   Académie                      3598 non-null   object 
 2   Code du département           3598 non-null   object 
 3   Département                   3598 non-null   object 
 4   UAI                           3598 non-null   object 
 5   Nom de l'établissment         3598 non-null   object 
 6   Code INSEE de la commune      3598 non-null   object 
 7   Nom de la commune             3598 non-null   object 
 8   Secteur                       3598 non-null   object 
 9   Type de lycée                 3598 non-null   object 
 10  Effectifs voie GT             2464 non-null   float64
 11  Effectifs voie PRO            2075 non-null   float64
 12  Effectifs Ensemble GT-PRO     3598 non-null   float64
 13  IPS

Le fichier fr-en-ips-lycees-ap2022.csv contient 18 colonnes<br>
<ul>
  <li>10 colonnes pour décrire le lycée (adresse, type de lycée,...)</li>
  <li>une colonne qui indique la rentrée au format XXXX-YYYY</li>
  <li>3 colonnes qui décrivent les effectifs général, professionnel et totaux</li>
  <li>5 colonnes pour décrire l'IPS selon le cursus général ou pro, ainsi que l'écart-type</li>
</ul>

On va créer un carnet d'adresses avec le numéro uai à partir de ce fichier.<br>
On va reformater la colonne rentrée scolaire. (passée de XXXX-YYYY à XXXX)

In [5]:
va_lycee_gen.head(1)

Etablissement  Annee   Ville       UAI  \
0  LYCEE XAVIER BICHAT (GENERAL ET TECHNO.)   2021  NANTUA  0010032E   

  Code commune Academie Departement Secteur  Presents - L  Presents - ES  ...  \
0        01269     LYON         AIN  public           NaN            NaN  ...   

   Taux de mentions - Gnle  Valeur ajoutée du taux de mentions - Gnle  \
0                     70.0                                         -7   

   Nombre de mentions TB avec félicitations - G  \
0                                           NaN   

   Nombre de mentions TB sans félicitations - G  Nombre de mentions B - G  \
0                                           NaN                       NaN   

   Nombre de mentions AB - G  Nombre de mentions TB avec félicitations - T  \
0                        NaN                                           NaN   

   Nombre de mentions TB sans félicitations - T  Nombre de mentions B - T  \
0                                           NaN                       NaN   

   Nombre de mentions AB - T  
0                        NaN  

[1 rows x 145 columns]

In [6]:
va_lycee_gen.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27808 entries, 0 to 27807
Columns: 145 entries, Etablissement to Nombre de mentions AB - T
dtypes: float64(80), int64(3), object(62)
memory usage: 30.8+ MB


Le fichier fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique.csv contient les 145 colonnes suivantes :

In [7]:
for a in va_lycee_gen.columns :
    print(a)

Etablissement
Annee
Ville
UAI
Code commune
Academie
Departement
Secteur
Presents - L
Presents - ES
Presents - S
Presents - STG
Presents - STI2D
Presents - STD2A
Presents - STMG
Presents - STI
Presents - STL
Presents - ST2S
Presents - TMD
Presents - STHR
Presents - Toutes series
Taux de reussite - L
Taux de reussite - ES
Taux de reussite - S
Taux de reussite - STG
Taux de reussite - STI2D
Taux de reussite - STD2A
Taux de reussite - STMG
Taux de reussite - STI
Taux de reussite - STL
Taux de reussite - ST2S
Taux de reussite - TMD
Taux de reussite - STHR
Taux de reussite - Toutes series
Taux de reussite attendu acad - L
Taux de reussite attendu acad - ES
Taux de reussite attendu acad - S
Taux de reussite attendu acad - STG
Taux de reussite attendu acad - STI2D
Taux de reussite attendu acad - STD2A
Taux de reussite attendu acad - STMG
Taux de reussite attendu acad - STI
Taux de reussite attendu acad - STL
Taux de reussite attendu acad - ST2S
Taux de reussite attendu acad - TMD
Taux de reuss

Le fichier fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique.csv contient 145 colonnes<br>
<ul>
  <li>une colonne annee qui est l'année de la rentrée scolaire</li>
  <li>11 colonne pour décrire l'établissement (adresse, secteur public ou privé, numéro uai)</li>
  <li>pour chaque serie:
    <ul>
      <li>une colonne effectif_present qui donne le nombre de candidats qui passe le bac</li>
      <li>le taux brut de réussite</li>
      <li>le taux de réussite attendu en France, dans l'académie</li>
      <li>le taux de mention brut et attendu</li>
      <li>la valeur ajoutée pour le taux de réussite et la valeur ajoutée pour le taux mention</li>
    </ul>
    </li>
    <li>8 colonnes qui donnent le nombre de mentions par collège toute séries confondus</li>
</ul>
Alors pour limiter le nombre de colonnes : création d'une colonne serie et fusion de toutes les colonnes des series.<br>


In [8]:
va_lycee_pro.head(1)

Etablissement       UAI  Annee             Ville  \
0  LYCEE PROFESSIONNEL SAINTE JEANNE D'ARC  0592928M   2019  AULNOYE AYMERIES   

  Code commune Academie Departement             Secteur  \
0        59033    LILLE        NORD  privé sous contrat   

   Presents - Specialites pluri-technologiques de la production  \
0                                                NaN              

   Presents - Transformations  ...  \
0                         5.0  ...   

   Valeur ajoutee du taux de mentions - Specialites plurivalentes des services  \
0                                                NaN                             

   Valeur ajoutee du taux de mentions - Echanges et gestion  \
0                                                NaN          

   Valeur ajoutee du taux de mentions - Communication et information  \
0                                                NaN                   

   Valeur ajoutee du taux de mentions - Services aux personnes  \
0                                                  3             

   Valeur ajoutee du taux de mentions - Services a la collectivite  \
0                                                NaN                 

   Valeur ajoutee du taux de mentions - Services  \
0                                              3   

   Valeur ajoutee du taux de mentions - Toutes series  \
0                                                5.0    

   Nombre de mentions TB sans félicitations - P  Nombre de mentions B - P  \
0                                           NaN                       NaN   

   Nombre de mentions AB - P  
0                        NaN  

[1 rows x 130 columns]

In [9]:
va_lycee_pro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24236 entries, 0 to 24235
Columns: 130 entries, Etablissement to Nombre de mentions AB - P
dtypes: float64(62), int64(3), object(65)
memory usage: 24.0+ MB


Le fichier fr-en-indicateurs-de-resultat-des-lycees-denseignement-professionnels.csv contient les 130 colonnes suivantes :

In [10]:
for a in va_lycee_pro.columns :
    print(a)

Etablissement
UAI
Annee
Ville
Code commune
Academie
Departement
Secteur
Presents - Specialites pluri-technologiques de la production
Presents - Transformations
Presents - Genie civil construction bois
Presents - Materiaux souples
Presents - Mecanique electricite electronique
Presents - Production
Presents - Specialites plurivalentes des services
Presents - Echanges et gestion
Presents - Communication et information
Presents - Services aux personnes
Presents - Services a la collectivite
Presents - Services
Presents - Toutes series
Taux de reussite - Specialites pluri-technologiques de la production
Taux de reussite - Transformations
Taux de reussite - Genie civil construction bois
Taux de reussite - Materiaux souples
Taux de reussite - Mecanique electricite electronique
Taux de reussite - Production
Taux de reussite - Specialites plurivalentes des services
Taux de reussite - Echanges et gestion
Taux de reussite - Communication et information
Taux de reussite - Services aux personnes
Tau

Le fichier contient les éléments suivants :<br>
<ul>
  <li>Une colonne annee avec l'année scolaire</li>
  <li>9 colonnes qui décrivent le lycée (adresse, secteur public ou privé, uai, ...)</li>
  <li>3 colonnes pour les mentions (pas de mention très bien avec félicitations par rapport au lycée général)</li>
  <li>pour chaque spécialité professionnelle :
    <ul>
      <li>une colonne effectif qui donne le nombre de candidats pour le diplôme preparé</li>
      <li>un taux brut de réussite</li>
      <li>un taux de réussite attendu en France</li>
      <li>un taux de réussite attendu dans l'académie</li>
      <li>un taux brut de mention</li>
      <li>un taux de mention attendu</li>
      <li>la valeur ajoutée due à l'établissement au taux de réussite</li>
      <li>la valeur ajoutée due à l'établissement au taux de mention</li>
      <li>la valeur ajoutée due à l'établissement au taux d'accès (le lycée garde ces élèves au cours de la formation.)</li>
    </ul>
  </li>
  <li>les taux d'accès de l'établissement</li>
</ul>

Pour l'instant, je me concentre sur les différentes series, et l'IPS des établissements.<br>
Pour cela, je vais créer comme pour le lycée général une colonne serie et fusionner les différentes colonnes communes.<br>

In [11]:
adresse_lycee = ips_lycee[['UAI','Académie','Département','Code du département',"Nom de l'établissment",'Code INSEE de la commune','Nom de la commune','Type de lycée']].groupby(by='UAI').apply(lambda x :x,include_groups=False)

In [12]:
adresse_lycee.head()

Académie Département Code du département  \
UAI                                                      
0010001W 682      LYON         AIN                 001   
0010006B 683      LYON         AIN                 001   
0010010F 2917     LYON         AIN                 001   
0010013J 684      LYON         AIN                 001   
0010014K 2156     LYON         AIN                 001   

                                           Nom de l'établissment  \
UAI                                                                
0010001W 682   LYCEE PROFESSIONNEL ALEXANDRE BERARD LYCEE DES...   
0010006B 683   LYCEE POLYVALENT SAINT EXUPERY LYCEE DES METIE...   
0010010F 2917            LYCEE GENERAL ET TECHNOLOGIQUE DU BUGEY   
0010013J 684                               LYCEE GENERAL LALANDE   
0010014K 2156        LYCEE GENERAL ET TECHNOLOGIQUE EDGAR QUINET   

              Code INSEE de la commune  Nom de la commune Type de lycée  
UAI                                                                      
0010001W 682                     01004  AMBERIEU EN BUGEY            LP  
0010006B 683                     01033         VALSERHONE           LPO  
0010010F 2917                    01034             BELLEY          LEGT  
0010013J 684                     01053    BOURG EN BRESSE          LEGT  
0010014K 2156                    01053    BOURG EN BRESSE          LEGT

Affichage des données uniques de la colonne Rentrée scolaire du fichier IPS :

In [13]:
print(ips_lycee['Rentrée scolaire'].unique())

['2022-2023']


Affichage des  éléments uniques après changement de la colonne Rentrée scolaire du fichier IPS :

In [14]:
ips_lycee['Rentrée scolaire'] = ips_lycee['Rentrée scolaire'].str[:4].astype(int)
print(ips_lycee['Rentrée scolaire'].unique())

[2022]


In [15]:
ips_lycee.drop(columns=['Secteur','Académie',
'Code du département','Département',"Nom de l'établissment",
'Code INSEE de la commune','Nom de la commune','Type de lycée',
'Effectifs voie GT','Effectifs voie PRO','Effectifs Ensemble GT-PRO'],inplace=True)

In [16]:
va_lycee_gen.rename(columns={'Annee':'Rentrée scolaire'},inplace=True)

In [17]:
# enlevé les resneignement type adresse
va_lycee_gen.drop(columns=['Code commune','Ville','Academie','Departement','Etablissement','Region','Libelle departement','Code region','Code departement'],inplace=True)

In [18]:
# faire une liste des series dans le fichier va_lycee_gen
liste_serie_gen = []

for nom_colonne in va_lycee_gen.columns:
    if nom_colonne.startswith('Presents'):
        liste_serie_gen.append(nom_colonne[8:])

In [19]:
# changer le "ND" par nan
va_lycee_gen.replace({'ND':np.nan},regex=True,inplace=True)

In [20]:
# ajouter des colonne vide
va_lycee_gen["Valeur ajoutee du taux de reussite - STG"] = np.nan
va_lycee_gen["Valeur ajoutee du taux de reussite - STI"] = np.nan
va_lycee_gen['Taux de reussite attendu france - Gnle'] = np.nan 


In [21]:
# faire un dataframe avec serie

lycee_gen = pd.DataFrame(columns=['Rentrée scolaire','UAI','Secteur','Serie','Type de diplome gen pro','Presents'
                                    ,'Taux de reussite','Taux de reussite attendu acad','Taux de reussite attendu france'
                                    ,'Taux de mentions','Taux de mentions attendu'
                                    ,"Valeur ajoutee du taux de reussite"
                                    ,"Valeur ajoutee du taux de mentions"])
for serie in  liste_serie_gen: 
    liste_colonne = []
    colonne_mention =[]
    dico_serie = {'Presents'+serie:'Presents'
                ,'Taux de reussite'+serie:'Taux de reussite'
                ,'Taux de reussite attendu acad'+serie:'Taux de reussite attendu acad'
                ,'Taux de reussite attendu france'+serie:'Taux de reussite attendu france'
                ,'Taux de mentions'+serie:'Taux de mentions'
                ,'Taux de mentions attendu'+serie:'Taux de mentions attendu'
                ,"Valeur ajoutee du taux de reussite"+serie:"Valeur ajoutee du taux de reussite"
                ,"Valeur ajoutee du taux de mentions"+serie:"Valeur ajoutee du taux de mentions"
                }

    for colonne in va_lycee_gen.columns:
        if colonne.endswith(serie):
            liste_colonne.append(colonne)
        if colonne.startswith('nombre_de_mentions'):
            colonne_mention.append(colonne)
    liste_colonne +=['Rentrée scolaire','Secteur','UAI']
    temps_data = va_lycee_gen[~va_lycee_gen['Presents'+serie].isna()][liste_colonne].copy()
    temps_data['Serie']= serie[2:]
    temps_data['Type de diplome gen pro'] = 'gen'
    temps_data.rename(columns=dico_serie,inplace=True)
    lycee_gen = pd.concat([lycee_gen.astype('object'),temps_data.astype('object')])


In [22]:
#changer les types

lycee_gen['Presents'] = lycee_gen['Presents'].astype(int)
lycee_gen['Taux de reussite'] = lycee_gen['Taux de reussite'].astype(float)
lycee_gen['Taux de reussite attendu acad'] = lycee_gen['Taux de reussite attendu acad'].astype(float)
lycee_gen['Taux de mentions'] = lycee_gen['Taux de mentions'].astype(float)
lycee_gen['Valeur ajoutee du taux de reussite'] = lycee_gen['Valeur ajoutee du taux de reussite'].astype(float)
lycee_gen['Valeur ajoutee du taux de mentions'] = lycee_gen['Valeur ajoutee du taux de mentions'].astype(float)
lycee_gen['Valeur ajoutee du taux de réussite - Gnle'] = lycee_gen['Valeur ajoutee du taux de réussite - Gnle'].astype(float)
lycee_gen['Valeur ajoutée du taux de mentions - Gnle'] = lycee_gen['Valeur ajoutée du taux de mentions - Gnle'].astype(float)

# des mauvaises valeur sont dans le dataframe
lycee_gen['Taux de mentions attendu'] = lycee_gen['Taux de mentions attendu'].replace({'56(1)':56,'36(1)':36})
lycee_gen['Taux de reussite attendu france'] = lycee_gen['Taux de reussite attendu france'].replace({'92(1)':92,'86(1)':86})

lycee_gen['Taux de mentions attendu'] = lycee_gen['Taux de mentions attendu'].astype(float)
lycee_gen['Taux de reussite attendu france'] = lycee_gen['Taux de reussite attendu france'].astype(float)


Le fichier lycee_gen après changement :

In [23]:
lycee_gen.info()

<class 'pandas.core.frame.DataFrame'>
Index: 120203 entries, 468 to 26317
Data columns (total 15 columns):
 #   Column                                     Non-Null Count   Dtype  
---  ------                                     --------------   -----  
 0   Rentrée scolaire                           120203 non-null  object 
 1   UAI                                        120203 non-null  object 
 2   Secteur                                    120203 non-null  object 
 3   Serie                                      120203 non-null  object 
 4   Type de diplome gen pro                    120203 non-null  object 
 5   Presents                                   120203 non-null  int64  
 6   Taux de reussite                           120203 non-null  float64
 7   Taux de reussite attendu acad              31561 non-null   float64
 8   Taux de reussite attendu france            85010 non-null   float64
 9   Taux de mentions                           66106 non-null   float64
 10  Taux de ment

In [24]:
va_lycee_pro.rename(columns={'Annee':'Rentrée scolaire'},inplace=True)

In [25]:
va_lycee_pro.drop(columns=['Etablissement','Ville','Code commune','Academie','Departement'],inplace=True)

In [26]:
#listez les serie générale et technologique
liste_serie_pro = []
for nom_colonne in va_lycee_pro.columns:
    if nom_colonne.startswith('Presents'):
        liste_serie_pro.append(nom_colonne[8:])

In [27]:
# faire un dataframe avec serie

lycee_pro = pd.DataFrame(columns=['Rentrée scolaire','UAI','Secteur','Serie','Type de diplome gen pro','Presents'
                                    ,'Taux de reussite','Taux de reussite attendu acad','Taux de reussite attendu france'
                                    ,'Taux de mentions','Taux de mentions attendu'
                                    ,'Valeur ajoutee du taux de reussite',"Valeur ajoutee du taux de mentions"])
for serie in  liste_serie_pro: 
    liste_colonne = []
    dico_serie = {'Presents'+serie:'Presents'
                ,'Taux de reussite'+serie:'Taux de reussite'
                ,'Taux de reussite attendu acad'+serie:'Taux de reussite attendu acad'
                ,'Taux de reussite attendu france'+serie:'Taux de reussite attendu france'
                ,'Taux de mentions'+serie:'Taux de mentions'
                ,'Taux de mentions attendu'+serie:'Taux de mentions attendu'
                ,'Valeur ajoutee du taux de reussite'+serie:'Valeur ajoutee du taux de reussite'
                ,"Valeur ajoutee du taux de mentions"+serie:"Valeur ajoutee du taux de mentions"
                }
    for colonne in va_lycee_pro.columns:
        if colonne.endswith(serie):
            liste_colonne.append(colonne)
    liste_colonne +=['Rentrée scolaire','Secteur','UAI']
    temps_data = va_lycee_pro[~va_lycee_pro['Presents'+serie].isna()][liste_colonne].copy()
    temps_data['Serie']= serie[2:]
    temps_data['Type de diplome gen pro'] = 'pro'
    temps_data.rename(columns=dico_serie,inplace=True)
    lycee_pro = pd.concat([lycee_pro.astype("object"),temps_data.astype('object')])

In [28]:
#changement des type de colonne
lycee_pro['Presents'] = lycee_pro['Presents'].astype(int)
lycee_pro['Taux de reussite'] = lycee_pro['Taux de reussite'].astype(float)
lycee_pro['Taux de reussite attendu acad'] = lycee_pro['Taux de reussite attendu acad'].astype(float)
lycee_pro['Taux de mentions'] = lycee_pro['Taux de mentions'].astype(float)


# erreurs dans la convertion : ND present remplacé par Nan et des valeur avec des XX(1)
lycee_pro['Taux de mentions attendu'] = lycee_pro['Taux de mentions attendu'].replace({'ND':np.nan})
lycee_pro['Taux de mentions attendu'] = lycee_pro['Taux de mentions attendu'].str[:2]
lycee_pro['Taux de mentions attendu'] = lycee_pro['Taux de mentions attendu'].astype(float)

lycee_pro['Valeur ajoutee du taux de mentions'] = lycee_pro['Valeur ajoutee du taux de mentions'].replace({'ND':np.nan})
lycee_pro['Valeur ajoutee du taux de mentions'] = lycee_pro['Valeur ajoutee du taux de mentions'].astype(float)

lycee_pro['Valeur ajoutee du taux de reussite'] = lycee_pro['Valeur ajoutee du taux de reussite'].replace({'ND':np.nan})
lycee_pro['Valeur ajoutee du taux de reussite'] = lycee_pro['Valeur ajoutee du taux de reussite'].astype(float)

lycee_pro['Taux de reussite attendu france'] = lycee_pro['Taux de reussite attendu france'].replace({'ND':np.nan})
lycee_pro['Taux de reussite attendu france'] = lycee_pro['Taux de reussite attendu france'].str[:2]
lycee_pro['Taux de reussite attendu france'] = lycee_pro['Taux de reussite attendu france'].astype(float)


Le Dataframe lycée_pro après changement :

In [29]:
lycee_pro.info()

<class 'pandas.core.frame.DataFrame'>
Index: 102793 entries, 36 to 24235
Data columns (total 13 columns):
 #   Column                              Non-Null Count   Dtype  
---  ------                              --------------   -----  
 0   Rentrée scolaire                    102793 non-null  object 
 1   UAI                                 102793 non-null  object 
 2   Secteur                             102793 non-null  object 
 3   Serie                               102793 non-null  object 
 4   Type de diplome gen pro             102793 non-null  object 
 5   Presents                            102793 non-null  int64  
 6   Taux de reussite                    102793 non-null  float64
 7   Taux de reussite attendu acad       8124 non-null    float64
 8   Taux de reussite attendu france     46236 non-null   float64
 9   Taux de mentions                    65827 non-null   float64
 10  Taux de mentions attendu            28914 non-null   float64
 11  Valeur ajoutee du taux de reuss

In [30]:
#merger les fichiers ensembles
fichier_lycee = pd.concat([lycee_gen,lycee_pro])
fichier_lycee = pd.merge(fichier_lycee,ips_lycee, on=['UAI','Rentrée scolaire'],how='left')
fichier_lycee = pd.merge(fichier_lycee,adresse_lycee, on='UAI',how='left')


In [31]:
fichier_lycee['Secteur'] = fichier_lycee['Secteur'].replace({'public':'PUBLIC','privé sous contrat':'PRIVE'})

Le fichier utilisé pour analyse :

In [32]:
fichier_lycee.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 222996 entries, 0 to 222995
Data columns (total 27 columns):
 #   Column                                     Non-Null Count   Dtype  
---  ------                                     --------------   -----  
 0   Rentrée scolaire                           222996 non-null  object 
 1   UAI                                        222996 non-null  object 
 2   Secteur                                    222996 non-null  object 
 3   Serie                                      222996 non-null  object 
 4   Type de diplome gen pro                    222996 non-null  object 
 5   Presents                                   222996 non-null  int64  
 6   Taux de reussite                           222996 non-null  float64
 7   Taux de reussite attendu acad              39685 non-null   float64
 8   Taux de reussite attendu france            131246 non-null  float64
 9   Taux de mentions                           131933 non-null  float64
 10  Taux de 

In [33]:
fichier_lycee.to_csv('donnee_temp/lycee.csv',index=False)